backend/src/common/model-provider/index.ts

```
@@ -0,0 +1,267 @@
+        responseType: 'stream',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      })
+      .subscribe({
+        next: (response) => {
+          let buffer = '';
+          response.data.on('data', (chunk: Buffer) => {
+            buffer += chunk.toString();
+            let newlineIndex;
+            while ((newlineIndex = buffer.indexOf('\n')) !== -1) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)